File: problems/0008_string-to-integer-atoi/one-loop/Solution.php
<?php
class Solution {
    /**
     * @param String $str
     * @return Integer
     */
    function myAtoi($str) {
        $sLen = strlen($str);
        if ($sLen == 0)
            return 0;
        $i = 0;
        for (; $i < $sLen && $str[$i] == ' '; ++$i);
        $sign = 1;
        if ($i < $sLen)
            if ($str[$i] == '-') {
                ++$i;
                $sign = -1;
            } else if ($str[$i] == '+')
                ++$i;
        $result = 0;
        for (; $i < $sLen && '0' <= $str[$i] && $str[$i] <= '9'; ++$i) {
            $result = $result * 10 + $sign * ($str[$i] - '0');
            if ($result < -pow(2, 31))
                return -pow(2, 31);
            else if ($result > pow(2, 31) - 1)
                return pow(2, 31) - 1;
        }
        return $result;
    }
}

File: problems/0008_string-to-integer-atoi/one-loop/solution.rb
#!/usr/bin/ruby

# @param {String} str
# @return {Integer}
def my_atoi(str)
    if str.length == 0
        return 0
    end
    i = 0
    while i < str.length and str[i] == ' '
        i += 1
    end
    sign = 1
    if i < str.length
        if str[i] == '-'
            i += 1
            sign = -1
        elsif str[i] == '+'
            i += 1
        end
    end
    result = 0
    while i < str.length and '0' <= str[i] and str[i] <= '9'
        result = result * 10 + sign * (str[i].ord - '0'.ord)
        if result < INT_MIN
            return INT_MIN
        elsif result > INT_MAX
            return INT_MAX
        end
        i += 1
    end
    return result
end

INT_MAX = 2**31 - 1
INT_MIN = -2**31
